cluster/engine.go

```diff
@@ -192,6 +196,27 @@ func (e *Engine) Status() string {
 	return "Unhealthy"
 }
 
+// IncFailureCount increases engine's failure count, and set engine as unhealthy if threshold is crossed
+func (e *Engine) IncFailureCount() {
```

Comment:
`incFailureCount` cannot be made private because dockerclient are used in `package swarm`.